Content.Tests/DMProject/Tests/Operators/equivalence.dm

```diff
@@ -14,3 +14,22 @@
 	ASSERT((l1 ~! l2) == TRUE)
 	ASSERT((l1 ~! l3) == TRUE)
 	ASSERT((l1 ~! l4) == TRUE)
+
+	var/matrix/m1 = matrix(1,2,3,4,5,6)
+	var/matrix/m2 = matrix(-1,-2,-3,-4,-5,6)
+	var/matrix/m3 = matrix(1,2,3,4,5,6)
+	m3.a = "poop"
+	ASSERT((m1 ~= m1) == TRUE)
+	ASSERT((m1 ~= m2) == FALSE)
+	ASSERT((m1 ~= m3) == FALSE)
+	ASSERT((m2 ~= m1) == FALSE)
+	ASSERT((m2 ~= m2) == TRUE)
+	ASSERT((m2 ~= m3) == FALSE)
+	ASSERT((m3 ~= m1) == FALSE)
+	ASSERT((m3 ~= m2) == FALSE)
+	ASSERT((m3 ~= m3) == TRUE)
```

Comment:
None of these are testing that two different matrices with the same values are considered equivalent. That should be added.